content/blog/rag-deployment/index.md

```diff
@@ -0,0 +1,147 @@
+
+  
+
+How you send the raw files to your external service depends largely on your application and the origin of your files. For example, you can write a script that sends batches of files to your indexing endpoint at a fixed time each day or week, or you can configure a stream that sends the files as they come in. When you expect new data to index, you can enable autoscaling in Kubernetes, which will cause it to create replicas of your indexing pipeline that can then run in parallel.
+
+  
+
+Because indexing as a process is not as time-sensitive as querying, you can queue your files for indexing. Queues are used to ensure that requests can be accepted almost immediately, but processed sometime in the (near) future. To learn more about queuing with KEDA, see our [series of articles about scaling indexing pipelines](https://haystack.deepset.ai/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/).
+
+  
+
+Once your document embeddings are ready, the service adds them to the database, where they can now be queried.
+
+## Deploying the Query Pipeline to Production
+
+Unlike indexing, querying is extremely time-sensitive. When your users query your RAG system, they expect answers quickly – even the slightest delay can cause them to give up on your product and look elsewhere. Therefore, it is critical that your production query pipeline is always available and able to handle even a large number of queries at once. This means, among other things, that the pipeline must be able to scale on demand.
+
+  
+
+Again, Kubernetes is our friend. It receives the requests through the [Haystack REST API](https://docs.haystack.deepset.ai/docs/rest_api) endpoint and creates replicas of the query pipeline when many requests come in at once. It also ensures that these pipeline replicas have the necessary hardware resources available when needed. The queries are processed by the pipelines and the retrieval-augmented, LLM-generated responses are returned to your application, where your users can interact with them.
+
+![Sketch of multiple replicas of the RAG pipeline, receiving many user queries and communicating with the database.](replicas.png)
+
+While the nitty-gritty technical details of scaling are handled by Kubernetes, we have the ability to tweak it based on the type of pipeline. To do this, it’s useful to think about the following questions:
+
+  
+
```

Comment:
And 1 question: What's the role of model deployment services here? Does it make sense to mention them? E.g.: If I go with Sagemaker, will sagemaker handle scaling model requests? That incurs a cost but maybe it's worth it?
@izzbizz @ArzelaAscoIi

Reply:
Yes. We might want to be a bit careful with not create confusion for the reader, but what do you think about adding a comment about: "Considering an additional model hosting services like sagemaker (or hf  inference) might be helpful to spearately scale model inference"

Reply:
not 100% sure about the wording